miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultRunService.java

```diff
@@ -528,148 +520,167 @@ public void setRunDao(RunStore runDao) {
-        // A human user has never change the health of this run, so we will.
-        run.setHealth(dto.getHealthType());
-        isMutated = true;
-      }
-    }
+    isMutated |= updateContainerFromNotification(target, user, laneCount, containerSerialNumber, sequencer);
+    isMutated |= updateHealthFromNotification(source, target, user);
 
-    final Predicate<SequencingParameters> filterParameters;
-    switch (dto.getPlatformType()) {
+    switch (source.getPlatformType()) {
     case ILLUMINA:
-      isMutated |= updateField(((IlluminaNotificationDto) dto).getCallCycle(), ((IlluminaRun) run).getCallCycle(),
-          v -> ((IlluminaRun) run).setCallCycle(v));
-      isMutated |= updateField(((IlluminaNotificationDto) dto).getImgCycle(), ((IlluminaRun) run).getImgCycle(),
-          v -> ((IlluminaRun) run).setImgCycle(v));
-      isMutated |= updateField(((IlluminaNotificationDto) dto).getNumCycles(), ((IlluminaRun) run).getNumCycles(),
-          v -> ((IlluminaRun) run).setNumCycles(v));
-      isMutated |= updateField(((IlluminaNotificationDto) dto).getScoreCycle(), ((IlluminaRun) run).getScoreCycle(),
-          v -> ((IlluminaRun) run).setScoreCycle(v));
-      isMutated |= updateField(dto.isPairedEndRun(), run.getPairedEnd(), run::setPairedEnd);
-      // TODO filter the parameters based on the properties
-      filterParameters = p -> true;
+      isMutated |= updateIlluminaRunFromNotification((IlluminaRun) source, (IlluminaRun) target);
       break;
     case IONTORRENT:
       // Nothing to do
-      filterParameters = p -> true;
       break;
     case LS454:
-      isMutated |= updateField(((LS454NotificationDto) dto).getCycles(), ((LS454Run) run).getCycles(),
-          v -> ((LS454Run) run).setCycles(v));
-      isMutated |= updateField(dto.isPairedEndRun(), run.getPairedEnd(), run::setPairedEnd);
-      filterParameters = p -> true;
+      isMutated |= updateField(((LS454Run) source).getCycles(), ((LS454Run) target).getCycles(),
+          v -> ((LS454Run) target).setCycles(v));
+      isMutated |= updateField(source.getPairedEnd(), target.getPairedEnd(), target::setPairedEnd);
       break;
     case OXFORDNANOPORE:
       throw new NotImplementedException();
     case PACBIO:
-      isMutated |= updateField(((PacBioNotificationDto) dto).getMovieDurationInSec(), ((PacBioRun) run).getMovieDuration(),
-          v -> ((PacBioRun) run).setMovieDuration(v));
-      filterParameters = p -> true;
+      isMutated |= updateField(((PacBioRun) source).getMovieDuration(), ((PacBioRun) target).getMovieDuration(),
+          v -> ((PacBioRun) target).setMovieDuration(v));
       break;
     case SOLID:
       // Nothing to do
-      filterParameters = p -> true;
       break;
     default:
       throw new NotImplementedException();
     }
 
+    isMutated |= updateSequencingParameters(target, user, filterParameters, sequencer);
+
+    if (isNew) {
+      setChangeDetails(target);
```

Comment:
`setChangeDetails` could be called at the start of the `save` method, in which case all other calls to `setChangeDetails` could be removed from this class